### PR TITLE
Add clerk webhook for user create

### PR DESCRIPTION
**Pull request by kensonhui**

### CODE REVIEW

File: apps/web/prisma/schema.prisma [hunk 1]
```diff
@@ -17,15 +17,15 @@ type AnnotatedPdfHighlights {
 }
 
 type AnnotatedPdfHighlightsComments {
-  emoji     String?
-  text      String?
+  emoji     String
+  text      String
```
Comment (LemonFace0309) [inline apps/web/prisma/schema.prisma:21]: yeah try not to override these types pls. I overrode them to be optional on purpose

File: apps/web/prisma/schema.prisma [hunk 2]
```diff
@@ -17,15 +17,15 @@ type AnnotatedPdfHighlights {
...
   timestamp DateTime @db.Date
   userId    String
 }
 
 type AnnotatedPdfHighlightsContent {
-  image String?
-  text  String?
+  image String
```
Comment (LemonFace0309) [inline apps/web/prisma/schema.prisma:27]: likewise for this and all of the above. They're like this to try to minimize type errors in our existing code that has yet to be consolidated

File: apps/web/src/app/api/webhooks/route.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    return new Response('Error occured', {
+      status: 400
+    })
+  }
+ 
+  // Get the ID and type
+  const { id } = evt.data;
+  const eventType = evt.type;
+
+  console.log(body)
```
Comment (LemonFace0309) [inline apps/web/src/app/api/webhooks/route.ts:55]: try not to leave in console log messages

File: apps/web/src/app/api/webhooks/route.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+ 
+  if (eventType === "user.created") {
+    const {first_name, last_name, id: clerk_id} = payload.data;
+    const email = payload.data.email_addresses[0].email_address;
+    let handle = `${first_name}-${last_name}`.toLowerCase()
+
+
```
Comment (LemonFace0309) [inline apps/web/src/app/api/webhooks/route.ts:62]: make sure to format file and remove excess spaces

File: apps/web/src/app/api/webhooks/route.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        email,
+        handle,
+        clerk_id: clerk_id,
+        follows: [],
+        followers: []
+      }
+    })
+  }
+ 
+  return new Response('', { status: 200 })
```
Comment (LemonFace0309) [inline apps/web/src/app/api/webhooks/route.ts:93]: why not return the user record?

**Review by LemonFace0309 — CHANGES_REQUESTED** [review]
great job! Please make the requested changes. Thanks for opening up a PR for this and adding comments. cc @eden-chan for transparency 

Feel free to make a follow up PR adding user to TRPC context.

File: apps/web/src/app/page.tsx [hunk 6]
```diff
@@ -6,35 +6,37 @@ import { SignIn, currentUser } from "@clerk/nextjs";
...
       </div>
     );
   }
+  // TODO: Map Clerk userid to mongodb id
+  const clerkUserEmail = clerkUser?.emailAddresses[0]?.emailAddress;
+  const user = await api.user.fetchUser({
+    email: clerkUserEmail,
+  });
+
+  console.log(user, clerkUserEmail);
```
Comment (LemonFace0309) [inline apps/web/src/app/page.tsx:24]: remove the console log!

**Review by LemonFace0309 — APPROVED** [review]
looks great, ty king. Please remove the console log before merging

File: apps/web/src/app/page.tsx (hunk 6 above)
Comment (kensonhui) [reply apps/web/src/app/page.tsx]: Great catch thank you!

**Pull request merged**